src/isotools/splice_graph.py

```diff
@@ -40,21 +40,23 @@ def __init__(self, transcripts, strand):
         # get the links
         start_idx = {node.start: i for i, node in enumerate(self._graph)}
         end_idx = {node.end: i for i, node in enumerate(self._graph)}
-        self._tss = [start_idx[e[0][0]] for e in transcripts]  # todo: this is missleading: on - strand this would not be the tss
-        self._pas = [end_idx[e[-1][1]] for e in transcripts]
-
-        for i, tr in enumerate(transcripts):
-            for j, e in enumerate(tr):
-                if start_idx[e[0]] < end_idx[e[1]]:  # psydojuctions within exon
-                    self._graph[start_idx[e[0]]].suc[i] = start_idx[e[0]] + 1
-                    self._graph[end_idx[e[1]]].pre[i] = end_idx[e[1]] - 1
-                    for node_idx in range(start_idx[e[0]] + 1, end_idx[e[1]]):
+        self._tss = [start_idx[exon[0][0]] for exon in transcripts]  # todo: this is missleading: on - strand this would not be the tss
+        self._pas = [end_idx[exon[-1][1]] for exon in transcripts]
+
+        for i, transcript in enumerate(transcripts):
+            for j, exon in enumerate(transcript):
+                # pseudojunctions within exon
+                if start_idx[exon[0]] < end_idx[exon[1]]:
+                    self._graph[start_idx[exon[0]]].suc[i] = start_idx[exon[0]] + 1
+                    self._graph[end_idx[exon[1]]].pre[i] = end_idx[exon[1]] - 1
```

Comment:
I think it might be help to add abbreviations `suc` and `pre` in L15 to remind that `suc` refers to `successors` and `pre` refers to `predesessors` in this script:
`successors (suc) and predesessors (pre)`